### PR TITLE
src: Add message embed suppression

### LINKED ISSUE #3627: Allow editing a message to surppress embeds
**Is your feature request related to a problem? Please describe.**
About 3-4 weeks ago ish the discord api documentation was updated, you can now edit a messages flags to suppress embeds
**Describe the ideal solution**
Allow Message#edit to go through without an error if you're editing the flags and the message hasnt been sent by the bot
**Describe alternatives you've considered**

**Additional context**
https://discordapp.com/developers/docs/resources/channel#edit-message

> Edit a previously sent message. The fields content, embed, and flags can be edited by the original message author. __Other users can only edit flags and only if they have the `MANAGE_MESSAGES` permission in the corresponding channel__. When specifying flags, ensure to include all previously set flags/bits in addition to ones that you are modifying. Only flags documented in the table below may be modified by users (unsupported flag changes are currently ignored without error).

### PR DESCRIPTION
**Pull request by vladfrangu**
**Please describe the changes this PR makes and why it should be merged:**

Keeping it simple, adds support for suppressing and unsuppressing message embeds. Closes #3627

## TL;DR

```js
// Suppresses embeds
await m.suppressEmbeds();

// Unsuppresses embeds
await m.suppressEmbeds(false);

// Also suppresses embeds
await m.edit({ flags: ['SUPPRESS_EMBEDS'] });

// Also unsuppresses embeds
// Note however that this method may cause other issues if Discord decides to add more flags that can be added (or removed) by the end user. 
// Message#suppressEmbeds handles it differently, adding (or removing) just the bits for SUPPRESS_EMBEDS
await m.edit({ flags: [] });
```

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/structures/APIMessage.js [hunk 1]
```diff
@@ -63,6 +64,16 @@ class APIMessage {
     return this.target instanceof User || this.target instanceof GuildMember;
   }
 
+  /**
+   * Whether or not the target is a message
+   * @type {boolean}
+   * @readonly
+   */
+  get isMessage() {
+    const Message = require('./Message');
```
Comment (chroventer) [inline src/structures/APIMessage.js]: You can replace the require-statement from here to the top of the file since not only it is considered good practice but the code looks clean and organized.

Comment (PyroTechniac) [reply src/structures/APIMessage.js]: Actually no, as that causes circular requires.

**Comment by Renari** [comment]
It seems kind of out of place to implement `isMessage` in the APIMessage layer when that's supposed to be type agnostic rather than putting that logic in the Message struct itself.

**Comment by appellation** [comment]
> 
> 
> It seems kind of out of place to implement `isMessage` in the APIMessage layer when that's supposed to be type agnostic, rather than putting that logic in the Message struct itself.

I agree, but we already do this elsewhere in this class so refactoring that behavior would be somewhat out of scope.
